File: Frontend/src/app/services/requests/generate-data.request.ts
import { DataGenerationRate } from "src/app/models/data-generation-rate.enum";
import { DataType } from "src/app/models/data-type.enum";

export class GenerateDataRequest {
    dataGenerationRate: DataGenerationRate;
    dataType: DataType;
    connectionId: string;

    constructor(rate: DataGenerationRate, dataType: DataType, connectionId: string) {
        this.dataGenerationRate = rate;
        this.dataType = dataType;
        this.connectionId = connectionId;
    }
}

File: Backend/RealTimeCharts.Application/Data/Requests/GenerateDataRequest.cs
using MediatR;
using OperationResult;
using RealTimeCharts.Shared.Enums;

namespace RealTimeCharts.Application.Data.Requests
{
    public class GenerateDataRequest : IRequest<Result>
    {
        public GenerateDataRequest(DataGenerationRate dataGenerationRate, DataType dataType, string connectionId)
        {
            DataGenerationRate = dataGenerationRate;
            DataType = dataType;
            ConnectionId = connectionId;
        }

        public DataGenerationRate DataGenerationRate { get; set; }
        public DataType DataType { get; set; }
        public string ConnectionId { get; set; }
    }
}
